app/controllers/application_controller/filter.rb

```diff
@@ -956,7 +959,11 @@ def adv_search_build(model)
         s = MiqSearch.find(@edit[:expression][:selected][:id])
         clear_default_search if s.search_key == "_hidden_"
       end
-      @edit.delete(:exp_token)                                          # Remove any existing atom being edited
+      unless @edit.nil?
```

Comment:
@himdel, @karelhala - I have a separate issue open for this - https://github.com/ManageIQ/manageiq-ui-classic/issues/1542